src/mbgl/gl/segment.hpp

```diff
@@ -8,19 +8,19 @@ namespace gl {
 class Segment {
 public:
     Segment(std::size_t vertexOffset_,
-            std::size_t primitiveOffset_,
+            std::size_t indexOffset_,
             std::size_t vertexLength_ = 0,
-            std::size_t primitiveLength_ = 0)
+            std::size_t indexLength_ = 0)
         : vertexOffset(vertexOffset_),
-          primitiveOffset(primitiveOffset_),
+          indexOffset(indexOffset_),
           vertexLength(vertexLength_),
-          primitiveLength(primitiveLength_) {}
+          indexLength(indexLength_) {}
 
     const std::size_t vertexOffset;
-    const std::size_t primitiveOffset;
+    const std::size_t indexOffset;
 
     std::size_t vertexLength;
-    std::size_t primitiveLength;
+    std::size_t indexLength;
```

Comment:
Fundamentally, a vertex/index buffer is a buffer of vertices/indexes, not a buffer of triangles, or lines, or whatever. Conceptually, that's how OpenGL works, and I found it easier to reason about things when I stopped trying to think in terms of buffers of primitives.

Implementation wise, it simplifies calculations like [this](https://github.com/mapbox/mapbox-gl-native/pull/6856/files#diff-617d0e0d315b4076d9fb82555ca6df0cL464).

It matches the internal implementation of the new `{Vertex,Index}Vector` -- I think if we were to have `std::vector<Triangle<Vertex>>`, the packing might come out wrong. You don't want to get in a situation where there might be extra padding every third vertex. `{Vertex,Index}Vector` still enforces filling the vector with the right sized groups though.